pallets/process-validation/src/restrictions.rs

```diff
@@ -608,4 +626,72 @@ mod tests {
         );
         assert!(!result);
     }
+
+    #[test]
+    fn sender_has_input_role_succeeds() {
+        let roles = BTreeMap::from_iter(vec![(Default::default(), 1)]);
+        let inputs = vec![ProcessIO {
+            roles: roles.clone(),
+            metadata: BTreeMap::new(),
+            parent_index: None,
+        }];
+        let result = validate_restriction::<u64, u32, u32, u64>(
+            Restriction::SenderHasInputRole {
+                index: 0,
+                role_key: Default::default(),
```

Comment:
you asked before about other role_keys. Note that the `R` in `validate_restriction` is a `u32` so that's your `role_key` type

Reply:
thanks, I've added a test to check failing on an incorrect role (that isn't the default role)